application/app/views/dataverse/datasets/_dataset_info.html.erb

```diff
@@ -2,10 +2,38 @@
   <div class="card-header">
     <div class="d-flex justify-content-between align-items-center gap-3 flex-wrap">
       <div class="flex-grow-1 d-flex flex-column">
-        <h2 class="mb-0 text-nowrap h5"><%= t('.header_dataset_text') %> <%= dataset.data.dataset_persistent_id %></h2>
+        <div class="d-flex justify-content-between align-items-center gap-2">
+          <h2 class="mb-0 text-nowrap h5">
+            <%= t('.header_dataset_text') %> <%= dataset.data.dataset_persistent_id %>
+          </h2>
+          <% if dataset.version.present? %>
+            <a class="d-flex align-items-center text-decoration-none text-secondary"
+               data-controller="utils--icon-toggle lazy-loader"
+               data-utils--icon-toggle-icon-on-value="bi-plus-square"
+               data-utils--icon-toggle-icon-off-value="bi-dash-square"
+               data-action="click->utils--icon-toggle#toggle click->lazy-loader#toggle"
+               data-lazy-loader-container-id-value="dataset-versions-content"
+               data-lazy-loader-url-value="<%= widgets_path(widget_path: 'dataverse_dataset_versions', dataverse_url: dataverse_url, persistent_id: dataset.data.dataset_persistent_id) %>"
+               data-lazy-loader-reload-on-toggle-value="false"
+               href="#"
+               role="button"
+               aria-expanded="false"
+               aria-controls="dataset-versions">
+              <i data-utils--icon-toggle-target="icon" class="bi bi-plus-square me-2 transition" aria-hidden="true"></i>
+              <span class="badge text-bg-info"><%= dataset.version %></span>
+            </a>
```

Comment:
(1) I would try to add the text `Version` before or after the `+` icon, to explain what this field is.

(2) I would clarify somehow that draft versions can be accessed if you add an API key to the repository. We can add a UI component that says "Using API key" when present or an "Add API key to access private data" link redirecting to the repositories page or allowing inline API key update without leaving the dataset page.  

(3) we can use a different color on the badge when the version is :draft

Reply:
> (1) I would try to add the text Version ...

I did add `version` on the first prototype, but visually it did not look good. This is the best Look and Feel that I was able to create.

> (2)  I would clarify somehow  ...

Yes, some user feedback is requrired. There is enough space in the row, I will play with options.

> (3) we can use a different color ...

Not sure, after playing with (2) will see how it looks